src/internal/restgateway/restgateway.go

```diff
@@ -0,0 +1,114 @@
+package restgateway
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/client"
+	"github.com/pachyderm/pachyderm/v2/src/internal/log"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+
+	pfsgw "github.com/pachyderm/pachyderm/v2/src/pfs"
+	ppsgw "github.com/pachyderm/pachyderm/v2/src/pps"
+
+	admingw "github.com/pachyderm/pachyderm/v2/src/admin"
+	authgw "github.com/pachyderm/pachyderm/v2/src/auth"
+	debuggw "github.com/pachyderm/pachyderm/v2/src/debug"
+	enterprisegw "github.com/pachyderm/pachyderm/v2/src/enterprise"
+	identitygw "github.com/pachyderm/pachyderm/v2/src/identity"
+	licensegw "github.com/pachyderm/pachyderm/v2/src/license"
+	proxygw "github.com/pachyderm/pachyderm/v2/src/proxy"
+	transactiongw "github.com/pachyderm/pachyderm/v2/src/transaction"
+	versiongw "github.com/pachyderm/pachyderm/v2/src/version/versionpb"
+	workergw "github.com/pachyderm/pachyderm/v2/src/worker"
+)
+
+func NewRestGatewayMux(pachClientFactory func(context.Context) *client.APIClient) http.Handler {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
Prefer pctx.WithCancel() to context.WithCancel().  Our version reports the callsite of the cancel() invocation via context.Cause(ctx).  

In this case, I assume that any of the I/O we do will block until it's done, so by the time defer cancel() is run, nothing will be waiting on the context, so we probably don't need our own cancel context here.  If you want to bound startup time, context.WithTimeout is fine, but we have other ways to bound the startup time (k8s startup probe / liveness probe), so it's probably not needed.  Up to you.